datalad/core/distributed/clone.py

```diff
@@ -799,7 +799,20 @@ def postclonecfg_annexdataset(ds, reckless, description=None):
                     # Deal with file:// scheme URLs as well as plain paths.
                     # If origin isn't local, we have nothing to do.
                     origin_git_path = Path(RI(origin_annex_url).localpath)
-                    if origin_git_path.name != '.git':
+
+                    # we are local; check for a bare repo first to not mess w/
+                    # the path
+                    # Note, that w/o support for bare repos in GitRepo we also
+                    # can't use ConfigManager ATM.
+                    from datalad.cmd import GitWitlessRunner, StdOutErrCapture
```

Comment:
That is not true, and tested to be such: https://github.com/datalad/datalad/blob/6cbb11a28d28fa1e1eb2eb5b40722dcb3873d9d1/datalad/tests/test_config.py#L524

Reply:
It is true. `GitRepo` can only "handle" a bare repo, that it is actively creating. It can't be instantiated on an existing one. And `ConfigManager` requires a repo instance.

Reply:
Just looked into fixing `GitRepo`/`ConfigManager` for it, since it seemed to work for creation. Conclusion: It doesn't the test you pointed to is misleading. Doesn't actually read a bare repo's config. Instead it pretends to by first writing to it and then read out that value (into and from the wrong location, though).

I'll give it shot, but not within this PR.

Reply:
See PR #4911